nav2_route/include/nav2_route/corner_smoothing.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2025, Polymath Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_ROUTE__CORNER_SMOOTHING_HPP_
+#define NAV2_ROUTE__CORNER_SMOOTHING_HPP_
+
+#include <vector>
+#include <cmath>
+
+#include "nav2_route/types.hpp"
+#include "nav2_route/utils.hpp"
+
+namespace nav2_route
+{
+class CornerArc
+{
+public:
+  /**
+   * @brief Constructor
+   */
+  CornerArc(
+    const Coordinates & start, const Coordinates & corner, const Coordinates & end,
+    float minimum_radius)
+  {
+    start_edge_length_ = hypotf(corner.x - start.x, corner.y - start.y);
+    end_edge_length_ = hypotf(end.x - corner.x, end.y - corner.y);
+
+    double angle = getAngleBetweenEdges(start, corner, end);
+
+    double tangent_length = minimum_radius / (std::tan(std::fabs(angle) / 2));
+
+    if (tangent_length < start_edge_length_ && tangent_length < end_edge_length_) {
+      std::vector<double> start_edge_unit_tangent =
+      {(start.x - corner.x) / start_edge_length_, (start.y - corner.y) / start_edge_length_};
+      std::vector<double> end_edge_unit_tangent =
+      {(end.x - corner.x) / end_edge_length_, (end.y - corner.y) / end_edge_length_};
+
+      double bisector_x = start_edge_unit_tangent[0] + end_edge_unit_tangent[0];
+      double bisector_y = start_edge_unit_tangent[1] + end_edge_unit_tangent[1];
+      double bisector_magnitude = std::sqrt(bisector_x * bisector_x + bisector_y * bisector_y);
+
+      std::vector<double> unit_bisector =
+      {bisector_x / bisector_magnitude, bisector_y / bisector_magnitude};
```

Comment:
Just validating: is there any chance that the bisectors magnitudes could be  0? If so, we should check for NaN / Inf and set the variables appropriately for a failed corner attempt